src/lightning/data/processing/utilities.py

```diff
@@ -0,0 +1,80 @@
+import io
+import os
+import urllib
+from contextlib import contextmanager
+from subprocess import Popen
+from typing import Any, Callable, Optional, Tuple
+
+from lightning.data.constants import _IS_IN_STUDIO
+
+
+def get_worker_rank() -> Optional[str]:
+    return os.getenv("DATA_OPTIMIZER_GLOBAL_RANK")
+
+
+def catch(func: Callable) -> Callable:
+    def _wrapper(*args: Any, **kwargs: Any) -> Tuple[Any, Optional[Exception]]:
+        try:
+            return func(*args, **kwargs), None
+        except Exception as e:
+            return None, e
+    return _wrapper
+
+# Credit to the https://github.com/rom1504/img2dataset Github repo
+# The code was taken from there. It has a MIT License.
+
+def make_request(
+    url: str,
+    timeout: int = 10,
+    user_agent_token: str = "pytorch-lightning",
+) -> io.BytesIO:
+    """Download an image with urllib."""
+    user_agent_string = "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:72.0) Gecko/20100101 Firefox/72.0"
+    if user_agent_token:
+        user_agent_string += f" (compatible; {user_agent_token}; +https://github.com/Lightning-AI/pytorch-lightning)"
+
+    with urllib.request.urlopen(urllib.request.Request(url, data=None, headers={"User-Agent": user_agent_string}), timeout=timeout) as r:  # noqa: E501, S310
+        img_stream = io.BytesIO(r.read())
+    return img_stream
+
+
+@contextmanager
+def optimize_dns_context(enable: bool) -> Any:
+    optimize_dns(enable)
+    try:
```

Comment:
shouldn't it be

```
yield
try
...
except
...
```

Reply:
No, we want to catch any failures to disable back the context.